### ISSUE TITLE
Issues in the decrement and last emitted value

### ISSUE THREAD

**Issue by davideas** [BugSplat-Git/ngx-animated-counter#22]
It never decrements if we don't multiply the new value by result of `positiveOrNegative()`:
Fix with this:
``` typescript
scan((acc: number, curr: number) => acc + params.increment * curr)
```
Also we must be sure last emitted value corresponds to the endValue
Fix with `endWith()` after `takeWhile()`:
``` typescript
takeWhile(...),
// Make sure that last emitted value is end value
endWith(params.end)
```

**Comment by bobbyg603** [comment]
Hey @xvr-dev, can you take a look at this one?

**Comment by bobbyg603** [comment]
Thanks for reporting this issue @davideas, a fix is available in 14.0.1
